meson.build

```diff
@@ -0,0 +1,790 @@
+project('geany', 'c', 'cpp', version: '1.38')
+
+cc = meson.get_compiler('c')
+cxx = meson.get_compiler('cpp')
+prefix = get_option('prefix')
+
+deps_in = [
+    ['glib-2.0',    '2.32'],
+    ['gmodule-2.0', '2.32'],
+    ['gtk+-3.0',    '3.0']
+]
+
+deps = []
+deps_for_pc = ''
+foreach dep : deps_in
+    deps += [dependency(dep[0], version: '>= ' + dep[1])]
+    deps_for_pc += ' ' + dep[0] + ' >= ' + dep[1]
+endforeach
+
+glib = deps[0]
+
+cdata = configuration_data()
+check_headers = [
+    'dlfcn.h',
+    'errno.h',
+    'fcntl.h',
+    'glob.h',
+    'inttypes.h',
+    'langinfo.h',
+    'libintl.h',
+    'limits.h',
+    'locale.h',
+    'memory.h',
+    'stdbool.h',
+    'stdint.h',
+    'stdlib.h',
+    'string.h',
+    'sys/stat.h',
+    'sys/time.h',
+    'sys/types.h',
+    'unistd.h',
+    'wchar.h',
+    'wctype.h',
+]
+
+check_functions = [
+    ['fnmatch', '#include <fnmatch.h>'],
+    ['ftruncate', '#include <unistd.h>'],
+    ['gnu_get_libc_version', '#include<gnu/libc-version.h>'],
+    ['isblank', '#include <ctype.h>'],
+    ['mbrtowc', '#include <wchar.h>'],
+    ['memcpy',  '#include <string.h>'],
+    ['mkstemp', '#include <stdlib.h>'],
+    ['regcomp', '#include <regex.h>'],
+    ['socket', '#include <sys/socket.h>'],
+    ['strerror', '#include <string.h>'],
+    ['strstr', '#include <string.h>'],
+    ['wcrtomb', '#include <wchar.h>'],
+    ['wcscoll', '#include <wchar.h>'],
+]
+
+foreach h : check_headers
+    define = 'HAVE_' + h.underscorify().to_upper()
+    if cc.has_header(h)
+		cdata.set(define, 1)
+	else
+		cdata.set(define, false)
```

Comment:
Several times you do `cdata.get()` on values that could be either int `1` or bool `false`, and that evaluates to a meson syntax error when trying to check the "truthiness" of an integer. It doesn't build by default for me.

Which systems was this tested on? Seems like a system that has none of: glibc, fnmatch, regcomp (all 3 need to evaluate to false for configuration to pass).

Reply:
I tested on Windows at last. I'll check.